### ISSUE TITLE
Adding the regularization term to loss function ?

### ISSUE THREAD

**Issue by ricupa** [tristandeleu/pytorch-structured-sparsity#2]
I understand the changes you have made in the optimizer related to the proximal operator, but I am confused why you have not added the l1/l2 norm (like eq. 2 in the paper) to the loss function? 
In the test_vgg.py file the loss function is just the cross entropy loss, with no regularization term. 
Isn't it necessary ? 

I am asking because, I want to make 'lambda' learnable(or trainable) and for that lambda should be in the loss term to calculate the gradients. for backprop.
